src/ERC20/implementation/ERC20WrapperLocked.sol

```diff
@@ -0,0 +1,267 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import {ERC20Wrapper, ERC20, IERC20, SafeERC20} from "openzeppelin-contracts/token/ERC20/extensions/ERC20Wrapper.sol";
+import {Ownable, Context} from "openzeppelin-contracts/access/Ownable.sol";
+import {EnumerableMap} from "openzeppelin-contracts/utils/structs/EnumerableMap.sol";
+import {EVCUtil} from "ethereum-vault-connector/utils/EVCUtil.sol";
+
+/// @title ERC20WrapperLocked
+/// @custom:security-contact security@euler.xyz
+/// @author Euler Labs (https://www.eulerlabs.com/)
+/// @notice A wrapper for locked ERC20 tokens that can be withdrawn as per the lock schedule.
+/// @dev Regular wrapping (`depositFor`), unwrapping (`withdrawTo`), and `transfer` are only supported for whitelisted
+/// callers. `transferFrom` is only supported for whitelisted `from` addresses. If the account balance is
+/// non-whitelisted, their tokens can only be withdrawn as per the lock schedule and the remainder of the amount is
+/// burned.
```

Comment:
I would add a quick note, something like: "The wrapped token is assumed to not attempt to re-enter this wrapper, and otherwise be well-behaved."